andela/ah-backend-prime
# #164798310 Users should be able to report an article

#### What does this PR do?
- Allows users to report articles that violate terms of authors'  heaven
#### Description of tasks to be completed
- Add a model to store reported articles
- Inform a user that their article has been reported and action is being taken
- Implement logic to prevent an author from reporting their own articles.
#### How should this be manually tested?
- Fetch this branch by running this command `git fetch origin ft-report-article-164798310` in the terminal
- Check into this branch by running `git checkout ft-report-article-164798310`
- Install dependencies by running `pip install -r requirements.txt`
- Run migrations by running `python manage.py migrate`
- Start the development server by running `python manage.py runserver`
- Make a `POST` request to `/api/v1/articles/<slug>/report/`
#### Sample request body
```
{ "violation": "This article is offensive" }
```
#### What are the relevant pivotal tracker stories?
[#164798310](https://www.pivotaltracker.com/story/show/164798310)

#### Screenshots
- Authenticated user reporting his own article
![image](https://user-images.githubusercontent.com/36891299/56269273-fda4bb00-60fb-11e9-9a01-fe93d215f96f.png)
- Authenticated user reporting succeeds
![image](https://user-images.githubusercontent.com/36891299/56269307-0bf2d700-60fc-11e9-91ee-4907b30acb63.png)

- Email sent to author of reported article by admin
![image](https://user-images.githubusercontent.com/36891299/56272475-6e030a80-6103-11e9-837d-ee9ce4ed3a7c.png)



[Finishes #164798310]
